pkg/kubectl/cmd/certificates.go

```diff
@@ -0,0 +1,168 @@
+	return modifyCertificateCondition(f, out, cmd, args, options, func(csr *certificates.CertificateSigningRequest) (*certificates.CertificateSigningRequest, string) {
+		var alreadyApproved bool
+		for _, c := range csr.Status.Conditions {
+			if c.Type == certificates.CertificateApproved {
+				alreadyApproved = true
+			}
+
+		}
+		if alreadyApproved {
+			return csr, "approved"
+		}
+		csr.Status.Conditions = append(csr.Status.Conditions, certificates.CertificateSigningRequestCondition{
+			Type:           certificates.CertificateApproved,
+			Reason:         "KubectlApprove",
+			Message:        "This csr was approved by kubectl alpha approve",
+			LastUpdateTime: unversioned.Now(),
+		})
+		return csr, "approved"
+	})
+}
+
+func NewCmdCertificateDeny(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	options := CertificateOptions{}
+	cmd := &cobra.Command{
+		Use:   "deny SUBCOMMAND",
+		Short: "deny a CSR.",
+		Long:  ``,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(RunCertificateDeny(f, out, cmd, args, &options))
+		},
+	}
+	kubectl.AddJsonFilenameFlag(cmd, &options.Filenames, "filenames of certificatesigningrequest objects to deny")
+	cmdutil.AddRecursiveFlag(cmd, &options.Recursive)
+	cmdutil.AddOutputFlagsForMutation(cmd)
+
+	return cmd
+}
+
+func RunCertificateDeny(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string, options *CertificateOptions) error {
+	return modifyCertificateCondition(f, out, cmd, args, options, func(csr *certificates.CertificateSigningRequest) (*certificates.CertificateSigningRequest, string) {
+		var alreadyDenied bool
+		for _, c := range csr.Status.Conditions {
+			if c.Type == certificates.CertificateDenied {
+				alreadyDenied = true
+			}
+
+		}
+		if alreadyDenied {
+			return csr, "denied"
+		}
+		csr.Status.Conditions = append(csr.Status.Conditions, certificates.CertificateSigningRequestCondition{
+			Type:           certificates.CertificateDenied,
+			Reason:         "KubectlDeny",
+			Message:        "This csr was approved by kubectl alpha certificate deny",
+			LastUpdateTime: unversioned.Now(),
+		})
+		return csr, "denied"
+	})
+}
+
+func modifyCertificateCondition(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string, options *CertificateOptions, modify func(csr *certificates.CertificateSigningRequest) (*certificates.CertificateSigningRequest, string)) error {
+	var found int
+	mapper, typer := f.Object(false)
+	r := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+		ContinueOnError().
+		FilenameParam(false, options.Recursive, options.Filenames...).
+		ResourceNames("certificatesigningrequest", args...).
+		RequireObject(true).
+		Flatten().
+		Latest().
+		Do()
+	err := r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+		csr := info.Object.(*certificates.CertificateSigningRequest)
+		csr, verb := modify(csr)
+		err = info.Client.Put().
```

Comment:
Should be using generated client - this directly exposes raw actions and the client abstracts that intentionally.  Also client is versioned.